### PR TITLE
Preferences

### PR DESCRIPTION
**Pull request by webdev03**
Screen Shot 2021-12-21 at 11 23 58 PM

### CODE REVIEW

**Comment by webdev03** [comment]
Screen Shot 2021-12-21 at 7 15 04 PM
Working dark mode! Can someone tell me how to tell the webpages Dark Mode preference?

**Comment by webdev03** [comment]
In the future, if you want to change dark mode styles, by default style for light mode, then add `dark:` to the start of your tailwind classes for dark mode.

**Comment by webdev03** [comment]
Should I make custom search engines too?

**Comment by webdev03** [comment]
I have added user agent changing.

File: main/main.js [hunk 1]
```diff
@@ -2,16 +2,18 @@
 const { app, BrowserWindow } = require("electron");
 const path = require("path");
 
+let mainWindow;
 function createWindow() {
 	// Create the browser window.
-	const mainWindow = new BrowserWindow({
+	mainWindow = new BrowserWindow({
```
Comment (webdev03) [inline main/main.js:8]: This might help with Garbage collection.

File: main/main.js [hunk 2]
```diff
@@ -2,16 +2,18 @@
...
 		width: 1024,
 		height: 768,
 		webPreferences: {
 			preload: path.join(__dirname, "preload.js"),
 			webviewTag: true,
 			nodeIntegration: true,
 		},
+		title: "Catalyst",
```
Comment (webdev03) [inline main/main.js:16]: Just a minor change to the window title.

File: src/functions.js [hunk 3]
```diff
@@ -10,3 +10,11 @@ function generateHashkey() {
...
 	);
 }
+const removeChildren = (parent) => {
+	while (parent.lastChild) {
+		parent.removeChild(parent.lastChild);
+	}
+};
+const getPackageJSON = async() => {
+  return await (await fetch("../package.json")).json();
+}
```
Comment (webdev03) [inline src/functions.js:20]: Arrow functions! :smile:

File: src/index.html [hunk 4]
```diff
@@ -25,26 +27,49 @@
...
-    <button id="forward" onClick="forward()" class="text-center p-2 bg-green-300 hover:bg-green-400 transition"><i
+    <button id="forward" onclick="forward()" class="text-center p-2 bg-green-300 hover:bg-green-400 transition"><i
         style="color: black;" class="fa fa-forward"></i></button>
-    <input id="searchbar" type="text" style="width: 90%" class="min-w-fit bg-gray-100 p-2"
+    <input id="searchbar" type="text" style="width: 90%" class="min-w-fit bg-gray-100 dark:text-gray-900 p-2"
       placeholder="Browse the Web" />
-    <button id="go" onClick="loadURL()" class="text-center p-2 bg-purple-300 hover:bg-purple-400 transition">Go</button>
-    <button id="reload" onClick="reload()" class="text-center p-2 bg-red-300 hover:bg-red-400 transition rounded-r"><i
+    <button id="go" onclick="loadURL()" class="text-center p-2 bg-purple-300 hover:bg-purple-400 transition">Go</button>
+    <button id="reload" onclick="reload()" class="text-center p-2 bg-red-300 hover:bg-red-400 transition"><i
```
Comment (webdev03) [inline src/index.html:40]: _Apparently_ my VSCode seems to like `onclick` more than `onClick` so I changed it, as I think in the HTML spec it is supposed to be `onclick`.

File: src/search.js [hunk 5]
```diff
@@ -1,13 +1,12 @@
 // protocols
 const protocols = ["https", "http"];
-const removeChildren = (parent) => {
-	while (parent.lastChild) {
-		parent.removeChild(parent.lastChild);
-	}
-};
```
Comment (webdev03) [inline src/search.js:7]: The `removeChildren` function was moved to `functions.js`

File: src/tabs.js [hunk 6]
```diff
@@ -6,7 +6,11 @@ let hasFavicon = {};
 /**
  * Creates a new tab
  */
-function createTab() {
+async function createTab() {
```
Comment (webdev03) [inline src/tabs.js:9]: It is async for Line 10

**Review by webdev03 — COMMENTED** [review]
Here are some comments I'd like you to answer later, to see your opinion.

**Comment by webdev03** [comment]
Now you can choose if you want DuckDuckGo Autocompletions!

**Comment by JaydenDev** [comment]
It's all fine go ahead

**Comment by JaydenDev** [comment]
ready to merge?

**Comment by JaydenDev** [comment]
@webdev03

**Comment by JaydenDev** [comment]
alright.. ready or not im mergin.

**Comment by JaydenDev** [comment]
works great!

**Comment by webdev03** [comment]
:smile: Any suggestions?

**Comment by webdev03** [comment]
Also, do you like the `{{version}}` UA helper? Should I add more of those?

**Comment by JaydenDev** [comment]
dont comment on closed PRs, go to the features discussion page.

**Comment by webdev03** [comment]
Uh.. okay.. I'm working on my own browser.

**Pull request merged**